BlueDBOnDisk/src/main/java/io/bluedb/disk/Blutils.java

```diff
@@ -67,6 +67,17 @@ public int compare(X x1, X x2) {
 		return results;
 	}
 
+	public static <X> int lastIndex(List<X> values, Predicate<X> condition) {

```

Comment:
The values passed in are ordered where I actually use this.  However, it was hard to communicate to any future user of this method that they must be ordered so instead of I just literally check the lastIndex matching the condition whether or not it's ordered.